nixos/modules/services/networking/bind.nix

```diff
@@ -151,11 +156,19 @@ in
       wantedBy = [ "multi-user.target" ];
 
       preStart = ''
+        mkdir -m 0755 -p /etc/bind
+        if ! [ -f "/etc/bind/rndc.key" ]; then
+          ${pkgs.bind.out}/sbin/rndc-confgen -r /dev/urandom -c /etc/bind/rndc.key -u ${bindUser} -a -A hmac-sha256 2>/dev/null
+        fi
+
         ${pkgs.coreutils}/bin/mkdir -p /var/run/named
         chown ${bindUser} /var/run/named
       '';
 
-      script = "${pkgs.bind.out}/sbin/named -u ${bindUser} ${optionalString cfg.ipv4Only "-4"} -c ${cfg.configFile} -f";
+      script  = "${pkgs.bind.out}/sbin/named -u ${bindUser} ${optionalString cfg.ipv4Only "-4"} -c ${cfg.configFile} -f";
+      reload  = "${pkgs.bind.out}/sbin/rndc -k '/etc/bind/rndc.key' reload";
```

Comment:
`man named` says that

> In routine operation, signals should not be used to control the nameserver; rndc should be used instead.

So, basically, signals _will_ work, but they are not _the right_ way of getting things done. As far as I understand the plan is to get rid of signals eventually (which is impossible, of course).